src/WikiParsing/Articles/Article.fs

```diff
@@ -140,20 +133,16 @@ let isLocked word =
 
 let isEditable = not << isLocked
 
-let getPartsOfSpeech word =
+let getPartsOfSpeech =
     let partsOfSpeech = [ "podstatné jméno"; "přídavné jméno"; "sloveso" ]
     let isPartOfSpeech s = partsOfSpeech |> Seq.contains s
 
-    let content = getContent word
-    if content |> hasPart "čeština"
-    then
-        content
-        |> getPart "čeština"
-        |> getParts
-        |> Seq.map fst
-        |> Seq.filter isPartOfSpeech
-    else
-        Seq.empty
+    getContent
+    >> getPart "čeština"
+    >> Option.map getParts
```

Comment:
Option.map of composition of those 3 functions seems nicer too me :-) but if you think this way is more readable then I'm not against it
`Option.map (getParts >> Seq.map fst >> Seq.filter isPartOfSpeech)`

Reply:
Yeah. good one, corrected